### PR TITLE
Allow tensor as iou_threshold parameter to tf.image.non_max_suppression.

### PR DESCRIPTION
**Pull request by xodus7**
I have a use case where dynamically changing the iou_threshold parameter on a frozen graph is required. I'm not 100% positive this doesn't violate the backwards compatibility rules, and it does make the iou_threshold a required parameter. Feedback from the maintainers would be appreciated.

### CODE REVIEW

**Comment by rmlarsen** [comment]
Since this changes the functionality of the op, I believe you have to do this by 

1) adding a new kernel named NonMaxSuppressionOpV2 
2) migrate all producers (e.g. Python wrappers) to use that kernel instead.

See this for more details:
https://www.tensorflow.org/programmers_guide/data_versions

**Comment by vrv** [comment]
(Though you have to wait 3 weeks after checking in the new Op/OpKernel before changing the python code).

**Comment by xodus7** [comment]
Ok, I moved my changes into a new NonMaxSuppressionV2Op as suggested.

File: tensorflow/core/kernels/non_max_suppression_op.cc [hunk 1]
```diff
@@ -89,6 +89,61 @@ static inline float ComputeIOU(typename TTypes<float, 2>::ConstTensor boxes,
   return intersection_area / (area_i + area_j - intersection_area);
 }
 
+static void DoNonMaxSuppressionOp(OpKernelContext* context,
```
Comment (rmlarsen) [inline tensorflow/core/kernels/non_max_suppression_op.cc]: Nit: Could you put this in anonymous namespace instead of making it static?

**Review by rmlarsen — CHANGES_REQUESTED** [review]
Looks good! Just one nit.

File: tensorflow/core/kernels/non_max_suppression_op.cc (hunk 1 above)
Comment (xodus7) [inline tensorflow/core/kernels/non_max_suppression_op.cc]: Done!

**Review by rmlarsen — APPROVED** [review]

**Comment by rmlarsen** [comment]
@tensorflow-jenkins test this please

**Pull request merged**